### PR TITLE
Bringing the deployment infrastructure up

### PR DESCRIPTION
**Pull request by amitkrout**
**What type of PR is this?**
/kind enhancement

**What does this PR do / why we need it**:

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: test/e2e/kamsuite/kamsuite.go [hunk 1]
```diff
@@ -170,3 +174,81 @@ func createRepository() error {
...
+
+	argocdPassword, err := argocdAPIServerPassword()
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(argocdServer)
+	fmt.Println(argocdPassword)
+
+	cmd := exec.Command(argocdPath, "login", "--username", "admin", "--password", argocdPassword, argocdServer, "--insecure")
```
Comment (amitkrout) [inline test/e2e/kamsuite/kamsuite.go:198]: Locally ```make e2e``` passes. 

Checking why it fails on CI

Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go:201]: This is the error hit in CI 

```INFO[2021-05-12T12:59:36Z] Error from server (NotFound): secrets "openshift-gitops-cluster" not found```

details - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-developer_kam/237/pull-ci-redhat-developer-kam-master-v4.7-integration-e2e/1392449843313512448#1:build-log.txt%3A371

Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go:201]: Pr https://github.com/redhat-developer/kam/pull/239 will fix it.

**Comment by amitkrout** [comment]
Platform issue 
/retest

File: test/e2e/kamsuite/kamsuite.go (hunk 1 above)
Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go]: Hit a new error 
``` ERRO[2021-05-12T17:31:50Z] time="2021-05-12T17:17:21Z" level=fatal msg=EOF```

Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go]: Now using ```--grpc-web``` flag in ```argocd login ...``` command

Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go]: The issue was coming due to the pod was not ready by the time i was trying to login argocd server.

```
 INFO[2021-05-13T09:25:33Z] cluster-b5798d6f9-lrf2p                                       1/1     Running             0          27s 
INFO[2021-05-13T09:25:33Z] kam-69866d7c48-mqh89                                          0/1     ContainerCreating   0          26s 
INFO[2021-05-13T09:25:33Z] openshift-gitops-application-controller-0                     0/1     ContainerCreating   0          24s 
INFO[2021-05-13T09:25:33Z] openshift-gitops-applicationset-controller-6447b8dfdd-n2hsv   0/1     ContainerCreating   0          24s 
INFO[2021-05-13T09:25:33Z] openshift-gitops-redis-74bd8d7d96-7xj4h                       0/1     ContainerCreating   0          24s 
INFO[2021-05-13T09:25:33Z] openshift-gitops-repo-server-c999f75d5-pz2mz                  0/1     ContainerCreating   0          24s 
INFO[2021-05-13T09:25:33Z] openshift-gitops-server-6ff4fbc8f6-blzfg                      0/1     ContainerCreating   0          24s 
```
Commit https://github.com/redhat-developer/kam/pull/237/commits/e9925b19a6a6b30ab0e02cc81643afbf146fff2d will wait upto 2 min for the pod running status.

Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go]: Still failing even after the pods are up and running 

```
 INFO[2021-05-14T08:33:34Z] cluster-b5798d6f9-42sgt                                       1/1     Running   0          38s 
INFO[2021-05-14T08:33:34Z] kam-69866d7c48-2m67m                                          1/1     Running   0          37s 
INFO[2021-05-14T08:33:34Z] openshift-gitops-application-controller-0                     0/1     Running   0          36s 
INFO[2021-05-14T08:33:34Z] openshift-gitops-applicationset-controller-6447b8dfdd-7m782   1/1     Running   0          35s 
INFO[2021-05-14T08:33:34Z] openshift-gitops-redis-74bd8d7d96-9jl8p                       1/1     Running   0          36s 
INFO[2021-05-14T08:33:34Z] openshift-gitops-repo-server-c999f75d5-m2rzq                  0/1     Running   0          36s 
INFO[2021-05-14T08:33:34Z] openshift-gitops-server-6ff4fbc8f6-t98jq                      0/1     Running   0          36s 
INFO[2021-05-14T08:33:34Z]                                              
INFO[2021-05-14T08:33:34Z]     Then login argocd API server                                                                                                                                                                                                                             # kamsuite.go:183 -> /kamsuite.loginArgoAPIServerLogin 
INFO[2021-05-14T08:33:34Z]     exit status 20
```

Adding a sleep of 60sec in the code

Comment (amitkrout) [reply test/e2e/kamsuite/kamsuite.go]: It works.

**Comment by amitkrout** [comment]
Pr https://github.com/redhat-developer/kam/pull/244 will fix bootstrap command failure

**Review by wtam2018 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**